src/Raven.Server/Documents/Handlers/Processors/Indexes/IndexHandlerProcessorForConvertAutoIndex.cs

```diff
@@ -0,0 +1,716 @@
+                    return;
+
+                sb.AppendLine($"Store(\"{fieldName}\", {typeof(FieldStorage).FullName}.{nameof(FieldStorage.Yes)});");
+            }
+
+            void HandleSuggestions(string fieldName, bool? suggestions)
+            {
+                if (suggestions.HasValue == false || suggestions == false)
+                    return;
+
+                sb.AppendLine($"Suggestion(x => x.{fieldName});");
+            }
+
+            void HandleSpatial(string fieldName, AutoSpatialOptions spatial, AutoIndexConversionContext context)
+            {
+                if (spatial == null)
+                    return;
+
+                var realFieldName = context.FieldNameMapping[fieldName];
+
+                sb.Append($"Spatial(\"{realFieldName}\", factory => factory.{spatial.Type}.");
+
+                switch (spatial.Type)
+                {
+                    case SpatialFieldType.Cartesian:
+
+                        switch (spatial.Strategy)
+                        {
+                            case SpatialSearchStrategy.QuadPrefixTree:
+                                sb.Append($"{nameof(CartesianSpatialOptionsFactory.QuadPrefixTreeIndex)}({spatial.MaxTreeLevel}, new {nameof(SpatialBounds)} {{ {nameof(SpatialBounds.MaxX)} = {spatial.MaxX}, {nameof(SpatialBounds.MaxY)} = {spatial.MaxY}, {nameof(SpatialBounds.MinX)} = {spatial.MinX}, {nameof(SpatialBounds.MinY)} = {spatial.MinY} }})");
+                                break;
+                            case SpatialSearchStrategy.BoundingBox:
+                                sb.Append($"{nameof(CartesianSpatialOptionsFactory.BoundingBoxIndex)}()");
+                                break;
+                            default:
+                                throw new ArgumentOutOfRangeException();
+                        }
+
+                        break;
+                    case SpatialFieldType.Geography:
+
+                        switch (spatial.Strategy)
+                        {
+                            case SpatialSearchStrategy.QuadPrefixTree:
+                                sb.Append($"{nameof(GeographySpatialOptionsFactory.QuadPrefixTreeIndex)}({spatial.MaxTreeLevel}, {typeof(SpatialUnits).FullName}.{spatial.Units})");
+                                break;
+                            case SpatialSearchStrategy.GeohashPrefixTree:
+                                sb.Append($"{nameof(GeographySpatialOptionsFactory.GeohashPrefixTreeIndex)}({spatial.MaxTreeLevel}, {typeof(SpatialUnits).FullName}.{spatial.Units})");
+                                break;
+                            case SpatialSearchStrategy.BoundingBox:
+                                sb.Append($"{nameof(GeographySpatialOptionsFactory.BoundingBoxIndex)}({typeof(SpatialUnits).FullName}.{spatial.Units})");
+                                break;
+                            default:
+                                throw new ArgumentOutOfRangeException();
```

Comment:
Please provide the argument name and the message to all `ArgumentOutOfRangeException`